### PR TITLE
Refine/dev review

### PR DESCRIPTION
**Pull request by Jeff-CCH**
- [x] remove --force in the git submodules
- [x] remove _ from the CallbackLibrary.sol
- [x] Add comment for Router's defaultCollector 
- [x] Add stackblitz urls for typehHash testing
- [x] refine FeeLibrary.t.sol & Router.t.sol
- [x] fix coverage stack too deep issue
- [x] fix readme grammar error
- [x] Apply multi-chain testing in forge

### CODE REVIEW

**Review by zodahu — APPROVED** [review]
LGTM

File: test/integration/fee/Permit2FeeCharging.t.sol [hunk 1]
```diff
@@ -37,6 +35,9 @@ contract Permit2FeeCalculatorTest is Test, ERC20Permit2Utils, TypedDataSignature
     bytes32[] public referralsEmpty;
 
     function setUp() external {
+        vm.createSelectFork(vm.rpcUrl('ethereum'));
+        TypedDataSignature.initialize();
```
Comment (hihiben) [inline test/integration/fee/Permit2FeeCharging.t.sol:39]: Is `TypedDataSignature` here necessary? Or we can simply use `initialize()`?

Comment (Jeff-CCH) [reply test/integration/fee/Permit2FeeCharging.t.sol:39]: The main idea is to know what are we going to `initialize` without browsing the code. I also tried another name like `initializeTypeDataSignature`, but it was not as clean as the one we used.

Comment (hihiben) [reply test/integration/fee/Permit2FeeCharging.t.sol:39]: Got it, makes sense.

**Review by hihiben — APPROVED** [review]
LGTM

**Pull request merged**